packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -54,6 +66,16 @@ export async function deployStack(options: DeployStackOptions): Promise<DeploySt
 
   const params = await prepareAssets(options.stack, options.toolkitInfo, options.reuseAssets);
 
+  // add passed CloudFormation parameters
+  for (const [paramName, paramValue] of Object.entries((options.parameters || {}))) {
+    if (paramValue) {
```

Comment:
We can consider looking at the existing stack here, and if there are any Parameters on the current stack that aren't passed here, we can be passing `UsePreviousValue: true`  to make ergonomics around parameters a lot better.

Maybe not for this PR, but keep that in mind.  

https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_Parameter.html

Reply:
Great idea. I think we can use it to make `--trust` better - see https://github.com/aws/aws-cdk/pull/5856#discussion_r376673418 .